Thinktecture.Relay.Server/Communication/RabbitMq/RabbitMqMessageDispatcher.cs

```diff
@@ -150,37 +133,40 @@ public void AcknowledgeRequest(string acknowledgeId)
 			}
 		}
 
-		public Task DispatchRequest(Guid linkId, IOnPremiseConnectorRequest request)
+		public void DispatchRequest(Guid linkId, IOnPremiseConnectorRequest request)
 		{
-			var content = _encoding.GetBytes(JsonConvert.SerializeObject(request));
-			var props = new BasicProperties()
-			{
-				ContentEncoding = "application/json",
-				DeliveryMode = 2,
-			};
+			var content = Serialize(request, out var props);
 
 			if (request.Expiration != TimeSpan.Zero)
 			{
 				_logger?.Verbose("Setting RabbitMQ message TTL. request-id={RequestId}, request-expiration={RequestExpiration}", request.RequestId, request.Expiration);
 				props.Expiration = request.Expiration.TotalMilliseconds.ToString(CultureInfo.InvariantCulture);
 			}
 
-			_model.BasicPublish(_EXCHANGE_NAME, linkId.ToString(), false, props, content);
+			_model.BasicPublish(_EXCHANGE_NAME, $"{_REQUEST_QUEUE_PREFIX}{linkId}", false, props, content);
+		}
+
+		public void DispatchResponse(Guid originId, IOnPremiseConnectorResponse response)
+		{
+			var content = Serialize(response, out var props);
+			_model.BasicPublish(_EXCHANGE_NAME, $"{_RESPONSE_QUEUE_PREFIX}{originId}", false, props, content);
+		}
 
-			return Task.CompletedTask;
+		public void DispatchAcknowledge(Guid originId, string acknowledgeId)
+		{
+			var content = Serialize(acknowledgeId, out var props);
+			_model.BasicPublish(_EXCHANGE_NAME, $"{_ACKNOWLEDGE_QUEUE_PREFIX}{originId}", false, props, content);
 		}
 
-		public Task DispatchResponse(Guid originId, IOnPremiseConnectorResponse response)
+		private byte[] Serialize<T>(T message, out BasicProperties props)
```

Comment:
Maybe consider returning a value tuple `(byte[] Content, BasicProperties Properties)` instead of using `out` for the second return value of this function.

Reply:
I don't like tuples, they have no names :)